docs/autogen.py

```diff
@@ -1,256 +1,41 @@
-                             [line[num_leading_spaces:]
-                             for line in snippet_lines[1:]])
-            # Most code snippets have 3 or 4 more leading spaces
-            # on inner lines, but not all. Remove them.
-            inner_lines = snippet_lines[1:-1]
-            leading_spaces = None
-            for line in inner_lines:
-                if not line or line[0] == '\n':
-                    continue
-                spaces = count_leading_spaces(line)
-                if leading_spaces is None:
-                    leading_spaces = spaces
-                if spaces < leading_spaces:
-                    leading_spaces = spaces
-            if leading_spaces:
-                snippet_lines = ([snippet_lines[0]] +
-                                 [line[leading_spaces:]
-                                  for line in snippet_lines[1:-1]] +
-                                 [snippet_lines[-1]])
-            snippet = '\n'.join(snippet_lines)
-            code_blocks.append(snippet)
-            tmp = tmp[index:]
-
-    # Format docstring lists.
-    section_regex = r'\n( +)# (.*)\n'
-    section_idx = re.search(section_regex, docstring)
-    shift = 0
-    sections = {}
-    while section_idx and section_idx.group(2):
-        anchor = section_idx.group(2)
-        leading_spaces = len(section_idx.group(1))
-        shift += section_idx.end()
-        next_section_idx = re.search(section_regex, docstring[shift:])
-        if next_section_idx is None:
-            section_end = -1
-        else:
-            section_end = shift + next_section_idx.start()
-        marker = '$' + anchor.replace(' ', '_') + '$'
-        docstring, content = process_list_block(docstring,
-                                                shift,
-                                                section_end,
-                                                leading_spaces,
-                                                marker)
-        sections[marker] = content
-        # `docstring` has changed, so we can't use `next_section_idx` anymore
-        # we have to recompute it
-        section_idx = re.search(section_regex, docstring[shift:])
-
-    # Format docstring section titles.
-    docstring = re.sub(r'\n(\s+)# (.*)\n',
-                       r'\n\1__\2__\n\n',
-                       docstring)
+class KerasDocumentationGenerator(DocumentationGenerator):
 
-    # Strip all remaining leading spaces.
-    lines = docstring.split('\n')
-    docstring = '\n'.join([line.lstrip(' ') for line in lines])
-
-    # Reinject list blocks.
-    for marker, content in sections.items():
-        docstring = docstring.replace(marker, content)
-
-    # Reinject code blocks.
-    for i, code_block in enumerate(code_blocks):
-        docstring = docstring.replace(
-            '$CODE_BLOCK_%d' % i, code_block)
-    return docstring
+    def process_function_docstring(self, docstring, function):
+        if ("keras.backend." in function.__module__
```

Comment:
Please use `'` as string delimiter everywhere for consistency.

Reply:
Sure, I modified it, but it seems that you have to approve the pull request again for me to be able to merge.